minion/understanding/api.py

```diff
@@ -35,21 +36,26 @@ class XMLParser(object):
 
 
 class APICommand(base.BaseCommand):
-    configuration = {
-        'method': 'get'
-    }
     threaded = True
 
+    @minion.core.utils.functions.configuration_getter
+    def get_url(self):
+        return ''
+
+    @minion.core.utils.functions.configuration_getter
+    def get_method(self):
+        return 'get'
+
     def _validate_configuration(self):
-        if not self.get_configuration('url'):
+        if not self.get_url():
```

Comment:
Can you look for a method called `RequiresNonEmptyConfiguration` or similar, it should basically do that